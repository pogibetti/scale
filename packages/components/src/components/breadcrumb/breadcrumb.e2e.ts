import { newE2EPage } from '@stencil/core/testing';

describe('scale-breadcrumb', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-breadcrumb/>');
    const element = await page.find('scale-breadcrumb');
    expect(element).toHaveClass('hydrated');
  });

  it('should set aria-label="Breadcrumb" on its nav element', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-breadcrumb/>');
    const element = await page.find('scale-breadcrumb >>> nav');
    expect(element.getAttribute('aria-label')).toBe('Breadcrumb');
  });

  it('should accept a custom separator via prop', async () => {
    const page = await newE2EPage();
    await page.setContent(`
      <scale-breadcrumb separator="/">
        <a href="/a">A</a>
        <a href="/b">B</a>
      </scale-breadcrumb>
    `);
    const element = await page.find('scale-breadcrumb >>> span');
    expect(element).toEqualText('/');
  });

  it('should accept a custom separator via slot', async () => {
    const page = await newE2EPage();
    await page.setContent(`
      <scale-breadcrumb separator="/">
        <span slot="separator">
          <span>/</span>
        </span>
        <a href="/a">A</a>
        <a href="/b">B</a>
      </scale-breadcrumb>
    `);
    const separator = await page.find('scale-breadcrumb >>> span');
    expect(separator.innerHTML.trim()).toBe('<span>/</span>');
    const list = await page.findAll('scale-breadcrumb >>> li');
    expect(list.length).toBe(2);
  });

  it('should add aria-current="page" attribute to last item, when link', async () => {
    const page = await newE2EPage();
    await page.setContent(`
      <scale-breadcrumb separator="/">
        <a href="/a">A</a>
        <a href="/b">B</a>
      </scale-breadcrumb>
    `);
    const element = await page.find('scale-breadcrumb >>> li:last-child a');
    expect(element.getAttribute('aria-current')).toBe('page');
  });

  it('should render last item as plain text when no href is found', async () => {
    const page = await newE2EPage();
    await page.setContent(`
      <scale-breadcrumb separator="/">
        <a href="/a">A</a>
        <span>B</span>
      </scale-breadcrumb>
    `);
    const element = await page.find('scale-breadcrumb >>> li:last-child span');
    expect(element).toEqualText('B');
  });
});
