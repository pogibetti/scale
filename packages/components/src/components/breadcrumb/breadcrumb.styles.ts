import { JssStyle } from 'jss';

export const styles: JssStyle = {
  breadcrumb: {
    marginLeft: '-0.4em', // TODO use theme
    '& ol': {
      listStyle: 'none',
      paddingLeft: 0,
      marginTop: 0,
      marginBottom: 0,
      display: 'flex',
    },
    '& li': {
      display: 'inline-flex',
      alignItems: 'center',
      marginRight: '0.125ch',
    },
  },
  separator: {
    display: 'flex',
    marginLeft: '0.125ch',
  },
  item: {
    padding: '0.4em', // TODO use theme
  },
  link: {
    textDecoration: 'none',
    padding: '0.4em', // TODO use theme
    borderRadius: 8, // TODO use theme
    border: '2px solid transparent', // TODO use theme (border-width)
    color: ({ colors }) => colors.text.default,
    '&:hover': {
      color: '#F90984', // TODO use theme
    },
    '&:active': {
      color: '#CB0068', // TODO use theme
    },
    '&:focus': {
      borderColor: '#009DE0', // TODO use theme
      outline: 'none',
    },
  },
  current: {
    fontWeight: ({ typography }) => typography.fontWeightBold,
  },
};
