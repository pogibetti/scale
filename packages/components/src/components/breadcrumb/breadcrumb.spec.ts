import { newSpecPage } from '@stencil/core/testing';
import { Breadcrumb } from './breadcrumb';
// import { styles } from './breadcrumb.styles';
// import jss from 'jss';

describe('Breadcrumb', () => {
  /* let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Breadcrumb();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  }); */

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Breadcrumb],
      html: `
        <scale-breadcrumb></scale-breadcrumb>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });
});
