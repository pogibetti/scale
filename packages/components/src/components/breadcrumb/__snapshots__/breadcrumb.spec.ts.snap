// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should match snapshot 1`] = `
<scale-breadcrumb>
  <mock:shadow-root>
    <style>
      .breadcrumb-0-0-1 {
  margin-left: -0.4em;
}
.breadcrumb-0-0-1 ol {
  display: flex;
  list-style: none;
  margin-top: 0;
  padding-left: 0;
  margin-bottom: 0;
}
.breadcrumb-0-0-1 li {
  display: inline-flex;
  align-items: center;
  margin-right: 0.125ch;
}
.separator-0-0-2 {
  display: flex;
  margin-left: 0.125ch;
}
.item-0-0-3 {
  padding: 0.4em;
}
.link-0-0-4 {
  border: 2px solid transparent;
  padding: 0.4em;
  border-radius: 8px;
  text-decoration: none;
  color: #000;
}
.link-0-0-4:hover {
  color: #F90984;
}
.link-0-0-4:active {
  color: #CB0068;
}
.link-0-0-4:focus {
  outline: none;
  border-color: #009DE0;
}
.current-0-0-5 {
  font-weight: 700;
}
    </style>
    <nav aria-label="Breadcrumb" class="breadcrumb-0-0-1">
      <ol></ol>
    </nav>
  </mock:shadow-root>
</scale-breadcrumb>
`;
